src/core/Table/filters/FilterWrapper.tsx

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import '@itwin/itwinui-css/css/table.css';
+import Popover from '../../utils/Popover';
+import { useTheme } from '../../utils/hooks/useTheme';
+import { ColumnInstance, HeaderGroup } from 'react-table';
+import { TableFilterValue } from './types';
+
+export type FilterWrapperProps<T extends Record<string, unknown>> = {
+  column: HeaderGroup<T>;
+  allColumns: ColumnInstance<T>[];
+  children: JSX.Element;
+  filterBody: React.ReactNode;
+  onShow: () => void;
+  onHide: () => void;
+};
+
+export const FilterWrapper = <T extends Record<string, unknown>>(
```

Comment:
Sure we need this as a component? It is just popover basically, if we take out placement logic... And the amount of components now is confusing to figure out, what does what.

Reply:
Refactored.